src/libsyncengine/update_detection/file_system_observer/snapshot/snapshotitem.h

```diff
@@ -20,6 +20,7 @@
 
 #include "libcommon/utility/types.h"
 #include "libcommonserver/utility/utility.h"
+#include "snapshot.h"
```

Comment:
```suggestion
```
It does not seem to be used.  
As `snapshot.h` already includes `snapshotitem.h`, we should avoid including `snapshot.h` in `snapshotitem.h`.

Reply:
It is needed by:
`friend bool Snapshot::path(const NodeId &, SyncPath &, bool &) const noexcept;`

snapshot.h doesn't include anymore snapshotitem.h but a forward definition:
`class SnapshotItem;`